From ebb65ff6498d81a846aa9d583b9a970562c3ae02 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Tue, 1 Oct 2024 11:30:18 +0200
Subject: [PATCH] man: using WantedBy=default.target is not a good idea

We had several users, that wrote their unit files with
WantedBy=default.target because it should be started "every time".
But for example in Fedora/CentOS/RHEL, this often breaks for
example selinux relabels (where we just want to do a relabel and reboot).

(cherry picked from commit 67b6404b80cf8078f3d9ec6d4c2f34ac25b15077)

Resolves: RHEL-36531
---
 man/systemd.special.xml | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/man/systemd.special.xml b/man/systemd.special.xml
index 1620895511..0c1f03ecf3 100644
--- a/man/systemd.special.xml
+++ b/man/systemd.special.xml
@@ -229,6 +229,11 @@
             names like <varname>single</varname>, <varname>rescue</varname>, <varname>1</varname>,
             <varname>3</varname>, <varname>5</varname>, …; see
             <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry>.</para>
+
+            <para>For typical unit files please set <literal>WantedBy=</literal> to a regular target (like
+            <filename>multi-user.target</filename> or <filename>graphical.target</filename>),
+            instead of <filename>default.target</filename>, since such a service will also be run on special
+            boots like on system update, emergency boot…</para>
           </listitem>
         </varlistentry>
         <varlistentry>
